MentalHospital.Tests/Tests.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Xunit;
+using Moq;
+using AutoMapper;
+using Shouldly;
+using MentalHospital.BLL.Mapper;
+using MentalHospital.BLL.Models;
+using MentalHospital.BLL.Services;
+using MentalHospital.DAL.Interfaces;
+using MentalHospital.DAL.Entities;
+
+
+namespace MentalHospital.Tests
+{
+    public class Tests
+    {
+        private readonly IMapper mapper = new Mapper(new MapperConfiguration(
+            cfg => cfg.AddProfile<MappingProfile>()));
+        private Mock<IPatientRepository> MockPatientRep = new Mock<IPatientRepository>();
+
+        private readonly string notAssignedMsg = "Method returned value is not assingable with expected type";
+        private readonly string isNullMsg = "Method returns null value";
+
+        [Fact]
+        public void CreaateReturnValue()
+        {
+            //Arrange
+            var pModel = InitialPatientModel();
+            MockPatientRep.Setup(pRep => pRep.Create(It.IsAny<Patient>()))
+                .Returns<Task<Patient>>(patient => patient);
+            var pService = new PatientService(MockPatientRep.Object, mapper);
+
+            //Act
+            var result = pService.Create(pModel);
+
+            //Assert
+            result.ShouldNotBeNull(isNullMsg);
+            result.ShouldBeAssignableTo<Task<PatientModel>>(notAssignedMsg);
```

Comment:
Bad way of asserting result, better to check fields in result instead of type
Also can be applied "Verify" method from moq